### ISSUE TITLE
Snort 3.1.19.0 config check randomly (?) fails to parse EXTERNAL_NET when defined as "!$HOME_NET"

### ISSUE THREAD

**Issue by turtlewax** [snort3/snort3#229]
/usr/local/snort/bin/snort -V
```
   ,,_     -*> Snort++ <*-
  o"  )~   Version 3.1.19.0
   ''''    By Martin Roesch & The Snort Team
           http://snort.org/contact#team
           Copyright (C) 2014-2021 Cisco and/or its affiliates. All rights reserved.
           Copyright (C) 1998-2013 Sourcefire, Inc., et al.
           Using DAQ version 3.0.5
           Using LuaJIT version 2.1.0-beta3
           Using OpenSSL 1.1.1l  24 Aug 2021
           Using libpcap version 1.10.1 (with TPACKET_V3)
           Using PCRE version 8.45 2021-06-15
           Using ZLIB version 1.2.11
           Using Hyperscan version 5.4.4 2021-10-30
           Using LZMA version 5.2.5
```
grep 'HOME_NET\\|EXTERNAL_NET' /usr/local/snort/etc/snort/snort.lua
```
-- HOME_NET and EXTERNAL_NET must be set now
HOME_NET = "[10.0.0.0/8 192.168.0.0/16 172.16.0.0/12]"
EXTERNAL_NET = "!$HOME_NET"
```

Now, let's just run a few checks one after another:

```
alpine [~]# /usr/local/snort/bin/snort -c /usr/local/snort/etc/snort/snort.lua -q
alpine [~]# /usr/local/snort/bin/snort -c /usr/local/snort/etc/snort/snort.lua -q
alpine [~]# /usr/local/snort/bin/snort -c /usr/local/snort/etc/snort/snort.lua -q
alpine [~]# /usr/local/snort/bin/snort -c /usr/local/snort/etc/snort/snort.lua -q
ERROR: /usr/local/snort/etc/snort/snort.lua: failed to parse the IP address: !$HOME_NET.
ERROR: snort3-app-detect.rules:21 undefined variable in the string: $EXTERNAL_NET.
ERROR: snort3-app-detect.rules:22 undefined variable in the string: $EXTERNAL_NET.
ERROR: snort3-app-detect.rules:23 Undefined variable in the string: $EXTERNAL_NET.
ERROR: snort3-app-detect.rules:24 Undefined variable in the string: $EXTERNAL_NET.
```

Hmm... on the fifth run it decided it couldn't parse, but the four previous (and several after) worked fine, until it happened again.. and again, and again.  Sometimes it happens once in ten runs, sometimes it happens ten times consecutively.  Race condition of some sort?

Happy to provide any further info if necessary, but it looks like this ought to be easy to reproduce.

**Comment by InverseRE** [comment]
We will take a look.

**Comment by InverseRE** [comment]
This will be fixed in 3.1.78

**Comment by turtlewax** [comment]
@InverseRE Thank you; I'm looking forward to this finally being corrected!

**Comment by turtlewax** [comment]
Happy to report that I have compiled and installed 3.1.78.0 (details below) and am so far unable to reproduce the issue.

```
alps [~]# /usr/local/snort/bin/snort -V

   ,,_     -*> Snort++ <*-
  o"  )~   Version 3.1.78.0
   ''''    By Martin Roesch & The Snort Team
           http://snort.org/contact#team
           Copyright (C) 2014-2024 Cisco and/or its affiliates. All rights reserved.
           Copyright (C) 1998-2013 Sourcefire, Inc., et al.
           Using DAQ version 3.0.14
           Using LuaJIT version 2.1.0-beta3
           Using OpenSSL 3.1.4 24 Oct 2023
           Using libpcap version 1.10.4 (with TPACKET_V3)
           Using PCRE version 8.45 2021-06-15
           Using ZLIB version 1.3
           Using Hyperscan version 5.4.11 2023-11-22
           Using LZMA version 5.4.5
```
Relevant snort.lua content:
```
---------------------------------------------------------------------------
-- 1. configure defaults
---------------------------------------------------------------------------

-- HOME_NET and EXTERNAL_NET must be set now
-- setup the network addresses you are protecting
HOME_NET = '[10.0.0.0/8 192.168.0.0/16 172.16.0.0/12]'

-- set up the external network addresses.
-- (leave as "any" in most situations)
EXTERNAL_NET = '!$HOME_NET'
```

Thank you!

**Comment by InverseRE** [comment]
The fix introduced in commit f19aa0b
